src/utils/internal/compose.ts

```diff
@@ -4,10 +4,10 @@ type LengthOfTuple<Tuple extends any[]> = Tuple extends { length: infer L }
   ? L
   : never
 
-type DropFirstInTuple<Tuple extends any[]> = Tuple extends [
+type DropFirstInTuple<Tuple extends any[]> = ((...args: Tuple) => any) extends (
   arg: any,
   ...rest: infer LastArg
-]
+) => any
   ? LastArg
   : Tuple
```

Comment:
Also I think that this should be `[]` here but I just wanted to fix the parsing problem, this might change behaviour (but I think this is a bug right now)
```suggestion
  : []
```

Reply:
We can address that in a separate pull request. I'd like to know more what bug this causes at the moment.